### PR TITLE
PIN-3598 Added getTenantBySelfcareId to public api

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala [hunk 1]
```diff
@@ -576,6 +576,24 @@ final case class TenantApiServiceImpl(
       .recover { case _: TenantNotFound => None }
   } yield tenant.map(_.toManagement)
 
+  override def getTenantBySelfcareId(selfcareId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerTenant: ToEntityMarshaller[Tenant]
+  ): Route = authorize(ADMIN_ROLE, API_ROLE, M2M_ROLE, SECURITY_ROLE, SUPPORT_ROLE) {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala]: if this route is required by the job (and to keep retrocompatibility with the management endpoint), an additional role is required
```suggestion
  ): Route = authorize(ADMIN_ROLE, API_ROLE, M2M_ROLE, SECURITY_ROLE, INTERNAL_ROLE, SUPPORT_ROLE) {
```

File: src/test/scala/it/pagopa/interop/tenantprocess/authz/TenantApiServiceAuthzSpec.scala [hunk 2]
```diff
@@ -58,6 +58,13 @@ class TenantApiServiceAuthzSpec extends ClusteredMUnitRouteTest with SpecData {
     )
   }
 
+  test("Tenant api should accept authorized roles for getTenant") {
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/tenantprocess/authz/TenantApiServiceAuthzSpec.scala]:
```suggestion
  test("Tenant api should accept authorized roles for getTenantBySelfcareId") {
```

File: src/main/scala/it/pagopa/interop/tenantprocess/service/impl/TenantManagementServiceImpl.scala [hunk 3]
```diff
@@ -108,6 +108,15 @@ final case class TenantManagementServiceImpl(
       .toFuture(TenantAttributeNotFound(tenantId, attributeId))
   } yield attribute
 
+  def getTenantBySelfcareId(selfcareId: UUID)(implicit contexts: Seq[(String, String)]): Future[Tenant] = withHeaders {
+    (bearerToken, correlationId, ip) =>
+      val request =
+        tenantApi.getTenantBySelfcareId(xCorrelationId = correlationId, selfcareId = selfcareId, xForwardedFor = ip)(
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/tenantprocess/service/impl/TenantManagementServiceImpl.scala]: I'm probably missing something, but I would expect this request to be done in the read model.

File: src/main/scala/it/pagopa/interop/tenantprocess/error/ResponseHandlers.scala [hunk 4]
```diff
@@ -158,6 +158,15 @@ object ResponseHandlers extends AkkaResponses {
       case Failure(ex)                                    => internalServerError(ex, logMessage)
     }
 
+  def getTenantBySelfcareIdResponse[T](logMessage: String)(
+    success: T => Route
+  )(result: Try[T])(implicit contexts: Seq[(String, String)], logger: LoggerTakingImplicit[ContextFieldsToLog]): Route =
+    result match {
+      case Success(s)                      => success(s)
+      case Failure(ex: TenantByIdNotFound) => notFound(ex, logMessage)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/tenantprocess/error/ResponseHandlers.scala]: this exception is not raised anywhere

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/tenantprocess/service/impl/TenantManagementServiceImpl.scala (hunk 3 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/tenantprocess/service/impl/TenantManagementServiceImpl.scala]: I am not able to find selfcareId into the tenants collections

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/tenantprocess/service/impl/TenantManagementServiceImpl.scala]: The `selfcareId` is an optional field and is populated the first time an institution enters the FE.
We should return `404` when a `selfcareId` request fetches no tenant.

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**